pkg/runtime/wfengine/activity.go

```diff
@@ -168,24 +168,30 @@ func (a *activityActor) executeActivity(ctx context.Context, actorID string, nam
 	wi.Properties[CallbackChannelProperty] = callback
 	if err = a.scheduler.ScheduleActivity(ctx, wi); err != nil {
 		if errors.Is(err, context.DeadlineExceeded) {
-			return newRecoverableError(errors.New(
-				"timed-out trying to schedule an activity execution - this can happen if too many activities are running in parallel or if the workflow engine isn't running"))
+			return newRecoverableError(fmt.Errorf("timed-out trying to schedule an activity execution - this can happen if too many activities are running in parallel or if the workflow engine isn't running: %w", err))
 		}
 		return newRecoverableError(fmt.Errorf("failed to schedule an activity execution: %w", err))
 	}
 
 loop:
 	for {
+		t := time.NewTimer(10 * time.Minute)
 		select {
 		case <-ctx.Done():
+			if !t.Stop() {
+				<-t.C
```

Comment:
Just curious: it's my understanding that `t.Stop()` returns false if the timer has already fired. One thing I didn't quite grasp is why we still need to read from the channel in such cases. Is it because leaving the "message" in the timer channel could cause a memory leak?

Reply:
That's my understanding too. (*Insert note on how Go can't have "leaks" in the proper sense (unless there's a goroutine leak), more about delayed collection by the GC*)

Reply:
In this case, the timer should not have fired because that's a separate `case` in the switch